packages/maskbook/src/plugins/ITO/Worker/apis/index.ts

```diff
@@ -48,17 +48,17 @@ const POOL_FIELDS = `
     }
 `
 
-async function fetchFromMarketSubgraph<T>(query: string, chainId?: ChainId) {
+async function fetchFromMarketSubgraph<T>(query: QueryOptions<{ [variable: string]: any }, any>, chainId?: ChainId) {
     const subgraphURL = getITOConstants(chainId ? chainId : currentChainIdSettings.value).SUBGRAPH_URL
     if (!subgraphURL) return null
-    const response = await fetch(subgraphURL, {
-        method: 'POST',
-        mode: 'cors',
-        body: stringify({ query }),
+
+    const apolloClient = new ApolloClient({
+        uri: subgraphURL,
+        cache: new InMemoryCache(),
```

Comment:
We can use `client.readQuery({ query })` to to cache query results after fetching them.